src/pubsub.js

```diff
@@ -313,9 +313,19 @@ class BasicPubSub extends Pubsub {
     })
 
     // Broadcast SUBSCRIBE to all peers
-    this.peers.forEach((peer) => {
-      peer.sendSubscriptions(newTopics)
-    })
+    this.peers.forEach((peer) => sendSubscriptionsOnceReady(peer))
+    // make sure that Gossipsub is already mounted
+    function sendSubscriptionsOnceReady (peer) {
```

Comment:
(same comment for `sendUnsubscriptionsOnceReady`)
Do we want this nested inside of `subscribe()` or can it be separated out? Alternatively, does it even need to be a function or can it be a code block within the `forEach`?

Reply:
Needs to be a function because it is recursive. `onConnection` we call the function once again.

In a first approach, I extracted and used a single function for both. But this implied that I pass the `peer.sendSubscriptions` / `peer.sendUnsubscriptions` functions via parameters, in order to know which function to pass. The code was not that readable with that approach, and I favored readability instead of less lines of code in this case. It is a total of 5 extra lines of code used, which seemed not a big deal to me.

Reply:
Ahhh I didn't see the recursion - makes sense.